test/tstQueryTreeCallbacks.cpp

```diff
@@ -75,18 +75,22 @@ BOOST_AUTO_TEST_CASE_TEMPLATE(callback_spatial_predicate, TreeTypeTraits,
   int const n = 10;
   Kokkos::View<ArborX::Point *, DeviceType> points(
       Kokkos::view_alloc(Kokkos::WithoutInitializing, "points"), n);
-  ArborX::Point const origin = {{0., 0., 0.}};
   Kokkos::parallel_for(
       Kokkos::RangePolicy<ExecutionSpace>(0, n), KOKKOS_LAMBDA(int i) {
         points(i) = {{(double)i, (double)i, (double)i}};
       });
-  auto points_host =
-      Kokkos::create_mirror_view_and_copy(Kokkos::HostSpace{}, points);
 
-  std::vector<Kokkos::pair<int, float>> values;
-  values.reserve(n);
-  for (int i = 0; i < n; ++i)
-    values.emplace_back(i, ArborX::Details::distance(points_host(i), origin));
+  Kokkos::View<Kokkos::pair<int, float> *, DeviceType> values_device(
+      "values_device", n);
+  Kokkos::parallel_for(
+      Kokkos::RangePolicy<ExecutionSpace>(0, n), KOKKOS_LAMBDA(int i) {
+        ArborX::Point const origin = {{0., 0., 0.}};
+        values_device(i) = {i, ArborX::Details::distance(points(i), origin)};
+      });
+  std::vector<Kokkos::pair<int, float>> values(n);
+  Kokkos::deep_copy(Kokkos::View<Kokkos::pair<int, float> *, Kokkos::HostSpace>(
+                        values.data(), n),
+                    values_device);
```

Comment:
How about encapsulating this into a helper function? Would also make the device side initiation only for SYCL, and use the standard one for everything else.

Here and the other 3 spots below.

Reply:
I'm not opposed to having a single path. What about a separate function, though? It seems that we could reduce the 4x duplication of the code with it.